### ISSUE TITLE
Jetpack Connection: welcome screen dismiss button is confusing for screen reader users

### ISSUE THREAD

**Issue by monsieur-z** [Automattic/jetpack#35480]
In the welcome screen, the dismiss button is read “Dismiss”. It’s confusing since SR users might not have noticed the banner. The accessible name (i.e., the text announced by SRs) should be more descriptive (e.g., “Don’t show the welcome message again”).

Issue reported in p1HpG7-qVW-p2.

<img width="1128" alt="Screenshot 2024-02-13 at 2 34 14 PM" src="https://github.com/Automattic/jetpack/assets/1620183/adc3497f-b067-4225-8fd0-0c177681bde9">

<img width="384" alt="Screenshot 2024-02-13 at 2 34 39 PM" src="https://github.com/Automattic/jetpack/assets/1620183/985777c4-b383-4c64-8b68-c122a29ee3f8">

---

- Adding an `aria-label` attribute on the `button` should be sufficient. Let's remove the `title` attribute, as the support from screen readers is less reliable.
